dasharo-stability/capsule-update.robot

```diff
@@ -70,6 +70,92 @@ CUP002.001 Capsule Update With Wrong GUID
     ${out}=    Execute UEFI Shell Command    CapsuleApp.efi -S
     Should Contain    ${out}    Capsule Status: Not Ready
 
+CUPXX3.001 Verifying UUID
+    [Documentation]    Check if UUID didn't change after Capsule Update.
+    Power On
+    Boot System Or From Connected Disk    ubuntu
+    Login To Linux
+    Switch To Root User
+    ${original_uuid}=    Get Firmware UUID
+    Log To Console    \n[before update] ${original_uuid}
+
+    Boot Into UEFI Shell
+    Perform Capsule Update    max_fw_ver.cap
+
+    Boot System Or From Connected Disk    ubuntu
+    Login To Linux
+    Switch To Root User
+    ${updated_uuid}=    Get Firmware UUID
+    Log To Console    \n[after update] ${updated_uuid}
+
+    Should Be Equal    ${original_uuid}    ${updated_uuid}
+
+    Restore DUT Initial State
+
+CUPXX4.001 Verifying Serial
+    [Documentation]    Check if UUID didn't change after Capsule Update.
+    Power On
+    Boot System Or From Connected Disk    ubuntu
+    Login To Linux
+    Switch To Root User
+    ${original_serial}=    Get Firmware Serial Number
+    Log To Console    \n[before update] ${original_serial}
+
+    Boot Into UEFI Shell
+    Perform Capsule Update    max_fw_ver.cap
+
+    Boot System Or From Connected Disk    ubuntu
+    Login To Linux
+    Switch To Root User
+    ${updated_serial}=    Get Firmware Serial Number
+    Log To Console    \n[after update] ${updated_serial}
+
+    Should Be Equal    ${original_serial}    ${updated_serial}
+
+    Restore DUT Initial State
+
+CUPXX5.001 Verifying If Custom Logo Persists Across updates
+    [Documentation]    Check if Logo didn't change after Capsule Update.
+    Skip If    not ${CUSTOM_LOGO_SUPPORT}    not supported
+
+    Prepare DUT For Logo Persistence Test
+
+    Boot Into UEFI Shell
+    Perform Capsule Update    max_fw_ver.cap
+
+    Boot System Or From Connected Disk    ubuntu
+    Login To Linux
+    Switch To Root User
+
+    ${out}=    Execute Command In Terminal    sha256sum /sys/firmware/acpi/bgrt/image
+
+    ${unplugged}=    Run Keyword And Return Status    Should Not Contain    ${out}    No such file
+
+CUPXX6.001 Verifying BIOS Settings Persistence After Update
+    [Documentation]    Check if BIOS settings didn't change after Capsule Update.
+    Power On
+    ${setup_menu}=    Enter Setup Menu Tianocore And Return Construction
+    ${dasharo_menu}=    Enter Dasharo System Features    ${setup_menu}
+    ${net_menu}=    Enter Dasharo Submenu    ${dasharo_menu}    Networking Options
+
+    Set Option State    ${net_menu}    Enable network boot    ${TRUE}
```

Comment:
This option is modified by many other tests, I don't think each of them restores it properly. Additionally, this is a boolean value, which means that with fully random value you have 50% chance of success. Perhaps setting [Auto Boot Time-out](https://docs.dasharo.com/dasharo-menu-docs/boot-maintenance-mgr/) to unreasonably high value and testing that would be better. It is also always present.

Reply:
If the goal is to test SMMSTORE preservation, I'd just add a generic EFI variable from Ubuntu that doesn't impact firmware functionality and isn't subject to resetting to a default. Use `efivar` from Ubuntu to create a new variable, update, and check if it's still there.

Reply:
We can test both of those cases. @mkopec could you suggest how to set or maybe what would be a good pick to use in such case?

Reply:
@krystian-hebel I have reworked this test to utilize Auto Boot Time-out: cdd63ce30ebcfe4732c50c4b8cfe834ed2923591

It is a 2 part test - again to update FW only once.

Reply:
This will be good for now. We may have to add test for cases like https://github.com/Dasharo/DasharoModulePkg/commit/1a488a4da5643010d32004ee41bba525bf84b44a in the future, but those will be added when such incompatibility appears.